backend/modules/cohorts/service.py

```
@@ -585,19 +595,62 @@ def assign_simulation_to_cohort(
+                    
+                    if existing_instance:
+                        # Reset existing instance when reassigning
+                        logger.info(f"Resetting existing instance {existing_instance.unique_id} for student {student.student_id} when reassigning simulation")
+                        
+                        # Delete existing user_progress if linked
+                        if existing_instance.user_progress_id and UserProgress:
+                            try:
+                                from modules.simulation.repository import SimulationRepository
+                                repo = SimulationRepository(self.db)
+                                repo.delete_user_progress_and_related(existing_instance.user_progress_id)
+                            except Exception as e:
```

Ruff (0.14.8)

Do not catch blind exception: `Exception`

(BLE001)